app/charts/shared/chart-state.ts

```diff
@@ -1,8 +1,23 @@
-import { ChartType } from "@/configurator";
+import { Bounds } from "@/charts/shared/use-width";
+import { ChartConfig, ChartType } from "@/configurator";
+import { Observation } from "@/domain/data";
 
-import { Bounds } from "./use-width";
+import { getAreasStateMetadata } from "../area/areas-state";
 
 export interface CommonChartState {
   chartType: ChartType;
   bounds: Bounds;
 }
+
+export type ChartStateMetadata = {
+  sortData: (data: Observation[]) => Observation[];
+};
+
+export const getChartStateMetadata = (
+  chartConfig: ChartConfig
+): ChartStateMetadata | undefined => {
+  switch (chartConfig.chartType) {
+    case "area":
+      return getAreasStateMetadata(chartConfig);
```

Comment:
By having a switch here, we might need to load all the getMetadata functions for all chart types (when we only need the getMetadata for 1 chart type, especially when viewing). I wonder if it would make sense to have a way so that the logic can be imported async.